proc/scc/scc.py

```diff
@@ -0,0 +1,703 @@
+import itertools
+import functools
+import time
+import math
+import copy
+
+import numpy as np
+
+import psi4
+import forte
+import forte.utils
+
+def run_cc(
+    forte_objs,
+    psi4_wfn,
+    cc_type=None,
+    select_type=None,
+    max_exc=None,
+    omega=None,
+    e_convergence=1.0e-10,
+    r_convergence=1.0e-5,
+    compute_threshold=1.0e-16,
+    selection_threshold=1.0e-16,
+):
+    """This function implements various selected CC methods
+
+    Parameters
+    ----------
+    forte_objs : tuple(ForteIntegrals, ActiveSpaceIntegrals, SCFInfo, MOSpaceInfo, map(StateInfo : list))
+        a tuple containing the ForteIntegrals, SCFInfo, and MOSpaceInfo objects and a map of states and weights
+    psi4_wfn : psi4 Wavefunction
+        A psi4 Wavefunction object (to read the number of alpha/beta electrons)
+    cc_type : str
+        The type of CC computation (cc/ucc/ducc)
+    select_type : str
+        The selection algorithm (default: None = no selection)
+    max_exc : int
+        The maximum excitation level (defaul: None)
+    omega : float
+        The selection threshold (default: None)
+    e_convergence : float
+        The energy convergence criterion (default = 1.0e-10)
+    r_convergence : float
+        The residual convergence criterion (default = =1.0e-5)
+    compute_threshold : float
+        The compute cutoff (default = =1.0e-16)
+    selection_threshold : float
+        The selection cutoff (default = =1.0e-16)
+    Returns
+    -------
+    list(tuple(int,float))
+        a list containing pairs of numbers that report the number of operators and the energy at each macroiteration
+    """
+
+    as_ints = forte_objs[1]
```

Comment:
1. I added the reference outputs. Thanks for catching this.
2. I think it might be able to run, CCSD/cc-pVTZ water, it just takes forever. I think with thresholding and selection things could be sped up by a lot, but this will be part of another PR.
3. I prefer to keep the jupyter notebooks without outputs to avoid merge conflicts. Unfortunately Jupyter does not play well with git and so minimal changes is the way to go.
4. BitArray is very similar to std::bitset (both pack bits into 64 unsigned longs), but implements efficient functions that require direct access to the  underlying data (e.g. bit counting, looping over bits)

Reply:
I agree that we use only part of `forte_objs`, but this is the only convenience function we have to prepare forte objects.

Reply:
@fevangelista 

3. That's fine. But even opening a Jupyter notebook may end up rewrite the file due to a different python version. Maybe we should add a README file to the tutorial directory and ask the user to create a subfolder and copy the notebooks there, unless the developer wants to change them.
4.  I think bitset supports bitwise operations and `count`. It may lack an iterator but `[]` works as intended. That's OK. I am just curious.